File: Task A/Tadas/taskA.cs
using System;

namespace Contestants.Tadas;

public static class Solver
{
    public static int? FirstUniqChar(string s)
    {
        // your code goes here
        return null;
    }

    public static void RunSmokeTests()
    {
        var cases = new (string input, int expected)[]
        {
            ("leetcode", 0),
            ("aabb", -1),
        };

        foreach (var (input, expected) in cases)
        {
            try
            {
                int? result = FirstUniqChar(input);
                Console.WriteLine($"{Format(input)} -> {result} (expected {expected})");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"{Format(input)} -> threw {ex.GetType().Name}: {ex.Message}");
            }
        }
    }

    private static string Format(string value) => value is null ? "null" : $"\"{value}\"";
}

File: Task A/Tadas/taskA.js
function firstUniqChar(s) {
  // your code goes here
}

function runSmokeTests() {
  const cases = [
    { input: "leetcode", expected: 0 },
    { input: "aabb", expected: -1 },
  ];

  console.log(`[Tadas] running smoke tests`);
  for (const { input, expected } of cases) {
    try {
      const result = firstUniqChar(input);
      console.log(`${JSON.stringify(input)} -> ${result} (expected ${expected})`);
    } catch (error) {
      const message = error instanceof Error ? error.message : String(error);
      console.log(`${JSON.stringify(input)} -> threw ${message}`);
    }
  }
}

module.exports = {
  firstUniqChar,
  runSmokeTests,
};

if (require.main === module) {
  runSmokeTests();
}
